### ISSUE TITLE
"Installing" pdf2htmlEX - Error: spawn pdf2htmlEX ENOENT or { code: 1 }

### ISSUE THREAD

**Issue by Jenack** [fagbokforlaget/pdftohtmljs#18]
Hi,

I'm trying to build a file uploader, which should take in only pdf's and uses pdf2htmlEX / your wrapper to convert the files to html.

I ran `npm install pdftohtmljs` and I now see **pdftohtmljs** in my node_modules folder.

However, for the life of me I cannot even get your usage code to work :(

```
var pdftohtml = require('pdftohtmljs');
var converter = new pdftohtml("/uploads/test.pdf", "upload/test.html");

converter.convert('ipad').then(function() {
  console.log("Success");
}).catch(function(err) {
  console.error("Conversion error: " + err);
});

// If you would like to tap into progress then create
// progress handler
converter.progress(function(ret) {
  console.log ((ret.current*100.0)/ret.total + " %");
});

```

I get this error if I don't have pdf2html "installed"

> events.js:183
>      throw er; // Unhandled 'error' event
>      ^
>
>Error: spawn pdf2htmlEX ENOENT
>    at _errnoException (util.js:1022:11)
>    at Process.ChildProcess._handle.onexit (internal/child_process.js:190:19)
>    at onErrorNT (internal/child_process.js:372:16)
>    at _combinedTickCallback (internal/process/next_tick.js:138:11)
>    at process._tickCallback (internal/process/next_tick.js:180:9)
>    at Function.Module.runMain (module.js:695:11)
>    at startup (bootstrap_node.js:188:16)
>    at bootstrap_node.js:609:3

And if I place pdf2thmlEX.exe in the root of my project I get this:

> Sorry, this script requires pdf2htmlEX. Please make sure its already installed.
> Install it from http://github.com/coolwanglu/pdf2htmlEX
> { code: 1 }

So basically my question is, how do I "install" pdf2thmlEX correctly, what am I doing wrong?

**Comment by iapain** [comment]
Hi,

May be error message was bit misleading, which should be fixed but error code: 1 explains that pdf2htmlEX.exe didn't run properly.

Instead of copying it to base folder, try to make it available through PATH variable. https://www.computerhope.com/issues/ch000549.htm

**Comment by iapain** [comment]
Implemented better error messages in v0.5.1

**Comment by shahzoobgouhar** [comment]
I am getting same error as mentioned by @Jenack . Tried solutions provided by @iapain but no luck. Can anyone provide more insights on how do I "install" pdf2thmlEX correctly" ?
